src/include/migraphx/op/gathernd.hpp

```diff
@@ -47,33 +48,136 @@ struct gathernd
 
     shape compute_shape(std::vector<shape> inputs) const
     {
-        check_shapes{inputs, *this}.has(2);
-        auto r = inputs.front().lens().size();
-        auto q = inputs.back().lens().size();
-        auto k = inputs.back().lens().back();
-        if(k > r - batch_dims)
+        check_shapes{inputs, *this, true}.has(2);
+        if(migraphx::none_of(inputs, [](auto v) { return v.dynamic(); }))
         {
-            MIGRAPHX_THROW("GATHERND: Indices of length " + std::to_string(k) +
-                           " cannot be used to access data of rank " +
-                           std::to_string(r - batch_dims));
+            auto r = inputs.front().ndim();
+            auto q = inputs.back().ndim();
+            auto k = inputs.back().lens().back();
+            if(k > r - batch_dims)
+            {
+                MIGRAPHX_THROW("GATHERND: Indices of length " + std::to_string(k) +
+                               " cannot be used to access data of rank " +
+                               std::to_string(r - batch_dims));
+            }
+            if(batch_dims >= q or batch_dims >= r)
+            {
+                MIGRAPHX_THROW("GATHERND: rank of an input cannot be less than batch_dims=" +
+                               std::to_string(batch_dims));
+            }
+            auto indices_lens_iter = inputs.back().lens().begin();
+            int output_lens_size   = int(q) + r - k - batch_dims - 1;
+            if(output_lens_size <= 0)
+            {
+                MIGRAPHX_THROW("GATHERND: Indices too large for static data input: k=" +
+                               std::to_string(k));
+            }
```

Comment:
Don't really need this since we already assert that `k > r -b` is NOT true. 
Therefore `1 <= k <= r-b` ---> `1 <= (r -b  -k)` and given `q >=1` ----> `output_lens_size >=1`.